packages/material-ui/src/FieldTemplate/FieldTemplate.tsx

```diff
@@ -43,22 +43,27 @@ const FieldTemplate = ({
         required={required}>
         {children}
         {displayLabel && rawDescription ? (
-          <Typography variant="caption" color="textSecondary">
+          <Typography
+            id={utils.descriptionId(id)}
+            variant="caption"
+            color="textSecondary">
             {rawDescription}
           </Typography>
         ) : null}
         {rawErrors.length > 0 && (
-          <List dense={true} disablePadding={true}>
+          <List id={utils.errorsId(id)} dense={true} disablePadding={true}>
             {rawErrors.map((error, i: number) => {
               return (
                 <ListItem key={i} disableGutters={true}>
-                  <FormHelperText id={id}>{error}</FormHelperText>
+                  <FormHelperText>{error}</FormHelperText>
```

Comment:
Why remove the id here?

Reply:
It was inside a `.map`. If there were multiple errors, it was not unique. Same id was also used with the `rawHelp` text. I moved the id to the parent `<List>` element and added suffix to both help and error list ids.

Reply:
Thanks! Can you add a short section in the "v3 upgrade guide" doc (within the documentation) that describes some of these breaking changes in id's? Just a quick list of what exactly were changed, so that someone who upgrades to v3 will know how the ids will change in the form.

Reply:
Could we consider this id change to be an internal fix? Element ids must be unique within the document and it is an undocumented feature. Here's a screenshot of the issue from the playground:
<img width="677" alt="Screenshot 2021-05-24 at 15 50 29" src="https://user-images.githubusercontent.com/5928960/119351770-88781c80-bca9-11eb-9da0-bad2730ee06d.png">

In this PR we let the input use the generated `id` (no change). To fix the case of non-unique ids and make `aria-describedby` behave correctly, we suffix help and errors ids (in material-ui FieldTemplate).